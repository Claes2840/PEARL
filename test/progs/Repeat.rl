(x n) -> (xs)
// xs is a list of n+1 x's

init: entry
      if n = '0 goto stop else loop

loop: fi xs = 'nil from init else loop
      xs <- (x . xs)
      x ^= hd xs
      n -= '1
      if n = '0 goto stop else loop

stop: fi xs = 'nil from init else loop
      xs <- (x . xs)
      n ^= '0
      exit