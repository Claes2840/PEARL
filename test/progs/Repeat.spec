list = 'nil
n = '3