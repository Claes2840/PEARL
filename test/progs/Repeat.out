(x) -> (xs)

init_1:
	entry
	assert((xs = 'nil))
	xs <- (x . xs)
	x ^= hd(xs)
	assert(!((xs = 'nil)))
	xs <- (x . xs)
	x ^= hd(xs)
	assert(!((xs = 'nil)))
	xs <- (x . xs)
	x ^= hd(xs)
	assert(!((xs = 'nil)))
	xs <- (x . xs)
	exit
